File: BAEKJOON/basic_algo_lecture/0x06_queue/10845.swift
var queue: [String] = []
var output: [String] = []

for _ in 1...Int(readLine()!)! {
	let command = readLine()!
	switch command {
	case "pop":
		if queue.isEmpty {
			output.append("-1")
		} else {
			output.append(queue.removeFirst())
		}
	case "size":
		output.append(String(queue.count))
	case "empty":
		if queue.isEmpty {
			output.append("1")
		}
		else {
			output.append("0")
		}
	case "front":
		if queue.isEmpty {
			output.append("-1")
		}
		else {
			output.append(queue.first!)
		}
	case "back":
		if queue.isEmpty {
			output.append("-1")
		}
		else {
			output.append(queue.last!)
		}
	default:
		let number = String(command.split(separator: " ")[1])
		queue.append(number)
	}
}

print(output.joined(separator: "\n"))

File: BAEKJOON/basic_algo_lecture/0x06_queue/10845.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
const N = input[0];
let queue = {}
let front = 0, back = -1, size = 0
let answer = ""

for (let i = 1; i <= N; i++) {
  const command = input[i];
  switch (command) {
    case "front":
        answer += size == 0 ? "-1\n" : `${queue[front]}\n`;
        break;
    case "back":
        answer += size == 0 ? "-1\n" : `${queue[back]}\n`;
        break;
    case "size":
        answer += `${size}\n`;
        break;
    case "pop":
        if (size == 0) {
            answer += "-1\n";
        } else {
            answer += `${queue[front]}\n`;
            front += 1;
            size -= 1;
        }
        break;  
    case "empty":
        answer += size == 0 ? "1\n" : "0\n";
        break;  
    default:
        const num = command.split(" ")[1];
        back += 1;
        queue[back] = num;
        size += 1;
        break;
  }
}

console.log(answer);